cmds.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+)
+
+// ErrUsage is a sentinel error for when commands detect an for invalid
+// combinations of flags or args. Usually kong handles all this, but sometimes
+// you cannot express an invalid combination of args/flags in the kong tags.
+// It will typically be wrapped so should be checked with `errors.Is()`.
+var ErrUsage = errors.New("usage error")
+
+// braviaAPI is a kong CLI struct to be embedded in command structs that
+// talk to a Sony Bravia TV set. It contains the parameters to communicate
+// with a TV using the Bravia REST IP control protocol.
+type braviaAPI struct {
+	Hostname string `env:"OFFSCREEN_HOSTNAME" help:"Hostname of Sony Bravia TV"`
+	PSK      string `env:"OFFSCREEN_PSK" help:"Pre-shared key"`
+}
+
+// SonyCmd is the kong CLI struct for the `sony` command.
+type SonyCmd struct {
+	Power  SonyCmdPower  `cmd:""`
+	Input  SonyCmdInput  `cmd:""`
+	Toggle SonyCmdToggle `cmd:""`
+
+	braviaAPI
+}
+
+// SonyCmdPower is the kong CLI struct for the `sony power` command.
+type SonyCmdPower struct {
+	State string `arg:"" optional:"" default:"" enum:",on,off" help:"Get/set power state"`
+}
+
+// SonyCmdInput is the kong CLI struct for the `sony input` command.
+type SonyCmdInput struct {
+	List  bool
+	Label string `arg:"" optional:"" default:"" help:"Get/set input"`
+}
+
+// SonyCmdToggle is the kong CLI struct for the `sony toggle` command.
+type SonyCmdToggle struct {
+	Input string `short:"i" help:"Specify host input, do not autodetect"`
+}
+
+// Run (sony power) gets or sets the power state of a Sony Bravia TV. If no
+// argument is provided, the current power state is printed. If the argument is
+// present and is "on", the TV is turned on. If it is "off" the TV is turned
+// off.
+func (sc *SonyCmdPower) Run(cli *CLI) error {
+	bc := NewRESTClient(cli.TV.Hostname, cli.TV.PSK)
```

Comment:
Hmmm, looks like I'm still using `bc` here and in the other commands. I'll need to rename these too.

Reply:
Done